src/apps/lwaftr/channel.lua

```diff
@@ -65,15 +65,16 @@ local function open_ring_buffer (pid, name)
    local path = root..'/'..tostring(pid)..'/channels/'..name
    local fd, err = S.open(path, "rdwr")
    if not fd then
-      local err = tostring(err or "unknown error")
+      err = tostring(err or "unknown error")
       error('error opening file "'..path..'": '..err)
    end
    local stat = S.fstat(fd)
    local len = stat and stat.size
    if len < ffi.sizeof(ring_buffer_t, 0) then
       error("unexpected size for ring buffer")
    end
-   local mem, err = S.mmap(nil, len, "read, write", "shared", fd, 0)
+   local mem
```

Comment:
there is nothing wrong with binding a fresh "err"; to my mind this is not a good change